File: language-speed-benchmarks/python/benchmark.py
import time

def is_prime(n):
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True

start = time.time()
count = sum(1 for i in range(2, 100000) if is_prime(i))
end = time.time()
print(f"Python: {end - start} seconds")

File: language-speed-benchmarks/rust/benchmark.rs
use std::time::Instant;

fn is_prime(n: u32) -> bool {
    if n <= 1 {
        return false;
    }
    for i in 2..=(n as f64).sqrt() as u32 {
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn main() {
    let start = Instant::now();
    let mut count = 0;
    for i in 2..100000 {
        if is_prime(i) {
            count += 1;
        }
    }
    let duration = start.elapsed();
    println!("Rust: {:.6} seconds", duration.as_secs_f64());
    println!("Prime count: {}", count);
}